src/megatron/bridge/training/checkpointing.py

```diff
@@ -137,6 +137,45 @@ def get_checkpoint_version() -> Optional[float]:
     return _CHECKPOINT_VERSION
 
 
+def delete_extra_state(state_dict):
+    """Delete all extra state keys from the model state dictionary.
+
+    This function removes all keys containing '_extra_state' from the model
+    portion of the state dictionary. This is useful for cleaning up corrupted
+    or problematic extra state that can cause issues during model loading.
+
+    Args:
+        state_dict: The state dictionary. Can be either:
+                   - A full checkpoint dict with a "model" key, or
+                   - A model state dict directly
+
+    Returns:
+        The modified state dictionary with extra state keys removed.
+    """
+    # Handle both cases: full checkpoint dict with "model" key or direct model state dict
+    if isinstance(state_dict, dict) and "model" in state_dict:
+        # Full checkpoint dict case
+        target_dict = state_dict["model"]
```

Comment:
this will miss keys formatted for pipeline parallelism: https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/6941a9319c2eaf297db7aa1252874c2f58ccfea5/src/megatron/bridge/training/checkpointing.py#L825-L836